### ISSUE TITLE
CCU Node error message in the log (IoBroker)

### ISSUE THREAD

**Issue by Schmello** [rdmtc/node-red-contrib-ccu#139]
Hi there...

I have installed NodeRed in the IOBroker and use the CCU Node.
This is set up. runs so far. But I keep getting the same red log entries.
Can someone explain to me what that can be?

```
node-red.0	2021-01-01 11:50:14.599	error	at CcuRpcNode.Node.emit (/opt/iobroker/node_modules/@node-red/runtime/lib/nodes/Node.js:179:25)
node-red.0	2021-01-01 11:50:14.599	error	at CcuRpcNode.Node._emitInput (/opt/iobroker/node_modules/@node-red/runtime/lib/nodes/Node.js:195:11)
node-red.0	2021-01-01 11:50:14.599	error	at Object.trigger (/opt/iobroker/node_modules/@node-red/runtime/lib/hooks.js:113:9)
node-red.0	2021-01-01 11:50:14.599	error	at /opt/iobroker/node_modules/@node-red/runtime/lib/nodes/Node.js:203:26
node-red.0	2021-01-01 11:50:14.599	error	at CcuRpcNode._inputCallback (/opt/iobroker/iobroker-data/node-red/node_modules/node-red-contrib-ccu/nodes/ccu-rpc.js:46:26)
node-red.0	2021-01-01 11:50:14.599	error	at CcuConnectionNode.methodCall (/opt/iobroker/iobroker-data/node-red/node_modules/node-red-contrib-ccu/nodes/ccu-connection.js:2673:20)
node-red.0	2021-01-01 11:50:14.599	error	at new Promise (<anonymous>)
node-red.0	2021-01-01 11:50:14.599	error	at /opt/iobroker/iobroker-data/node-red/node_modules/node-red-contrib-ccu/nodes/ccu-connection.js:2676:41
node-red.0	2021-01-01 11:50:14.599	error	at Client.methodCall (/opt/iobroker/iobroker-data/node-red/node_modules/homematic-xmlrpc/lib/client.js:100:30)
node-red.0	2021-01-01 11:50:14.599	error	at Object.exports.serializeMethodCall (/opt/iobroker/iobroker-data/node-red/node_modules/homematic-xmlrpc/lib/serializer.js:25:10)
node-red.0	2021-01-01 11:50:14.599	error	(13162) 1 Jan 11:50:14 - [error] [ccu-rpc:ReGaHSS RPC] TypeError: params.forEach is not a function
node-red.0	2021-01-01 11:48:13.989	error	at CcuRpcNode.Node.emit (/opt/iobroker/node_modules/@node-red/runtime/lib/nodes/Node.js:179:25)
node-red.0	2021-01-01 11:48:13.989	error	at CcuRpcNode.Node._emitInput (/opt/iobroker/node_modules/@node-red/runtime/lib/nodes/Node.js:195:11)
node-red.0	2021-01-01 11:48:13.989	error	at Object.trigger (/opt/iobroker/node_modules/@node-red/runtime/lib/hooks.js:113:9)
node-red.0	2021-01-01 11:48:13.989	error	at /opt/iobroker/node_modules/@node-red/runtime/lib/nodes/Node.js:203:26
node-red.0	2021-01-01 11:48:13.989	error	at CcuRpcNode._inputCallback (/opt/iobroker/iobroker-data/node-red/node_modules/node-red-contrib-ccu/nodes/ccu-rpc.js:46:26)
node-red.0	2021-01-01 11:48:13.989	error	at CcuConnectionNode.methodCall (/opt/iobroker/iobroker-data/node-red/node_modules/node-red-contrib-ccu/nodes/ccu-connection.js:2673:20)
node-red.0	2021-01-01 11:48:13.989	error	at new Promise (<anonymous>)
node-red.0	2021-01-01 11:48:13.989	error	at /opt/iobroker/iobroker-data/node-red/node_modules/node-red-contrib-ccu/nodes/ccu-connection.js:2676:41
node-red.0	2021-01-01 11:48:13.989	error	at Client.methodCall (/opt/iobroker/iobroker-data/node-red/node_modules/homematic-xmlrpc/lib/client.js:100:30)
node-red.0	2021-01-01 11:48:13.989	error	at Object.exports.serializeMethodCall (/opt/iobroker/iobroker-data/node-red/node_modules/homematic-xmlrpc/lib/serializer.js:25:10)
node-red.0	2021-01-01 11:48:13.989	error	(13162) 1 Jan 11:48:13 - [error] [ccu-rpc:ReGaHSS RPC] TypeError: params.forEach is not a function
```

**Comment by Sineos** [comment]
Are you using the IoBroker HM adapters in parallel?

**Comment by Schmello** [comment]
Yes. In Node Red i only must use the Interfaces ReGaHSS

**Comment by Sineos** [comment]
Not sure if this is going to work. Never tried.
Simply try deactivating all IoBroker HM Adapters and see if the error goes away.

**Comment by Schmello** [comment]
With disable all HM Adapter is the same...

```
node-red.0	2021-01-10 10:26:30.584	error	at CcuRpcNode.Node.emit (/opt/iobroker/node_modules/@node-red/runtime/lib/nodes/Node.js:179:25)
node-red.0	2021-01-10 10:26:30.584	error	at CcuRpcNode.Node._emitInput (/opt/iobroker/node_modules/@node-red/runtime/lib/nodes/Node.js:195:11)
node-red.0	2021-01-10 10:26:30.584	error	at Object.trigger (/opt/iobroker/node_modules/@node-red/runtime/lib/hooks.js:113:9)
node-red.0	2021-01-10 10:26:30.584	error	at /opt/iobroker/node_modules/@node-red/runtime/lib/nodes/Node.js:203:26
node-red.0	2021-01-10 10:26:30.584	error	at CcuRpcNode._inputCallback (/opt/iobroker/iobroker-data/node-red/node_modules/node-red-contrib-ccu/nodes/ccu-rpc.js:46:26)
node-red.0	2021-01-10 10:26:30.584	error	at CcuConnectionNode.methodCall (/opt/iobroker/iobroker-data/node-red/node_modules/node-red-contrib-ccu/nodes/ccu-connection.js:2673:20)
node-red.0	2021-01-10 10:26:30.584	error	at new Promise (<anonymous>)
node-red.0	2021-01-10 10:26:30.584	error	at /opt/iobroker/iobroker-data/node-red/node_modules/node-red-contrib-ccu/nodes/ccu-connection.js:2676:41
node-red.0	2021-01-10 10:26:30.584	error	at Client.methodCall (/opt/iobroker/iobroker-data/node-red/node_modules/homematic-xmlrpc/lib/client.js:100:30)
node-red.0	2021-01-10 10:26:30.584	error	at Object.exports.serializeMethodCall (/opt/iobroker/iobroker-data/node-red/node_modules/homematic-xmlrpc/lib/serializer.js:25:10)
node-red.0	2021-01-10 10:26:30.584	error	(31859) 10 Jan 10:26:30 - [error] [ccu-rpc:ReGaHSS RPC] TypeError: params.forEach is not a function

```

**Comment by hobbyquaker** [comment]
This log shows an RPC call failing and the error being reported through the rpc node — in the IoBroker-embedded setup of 2021 the root cause (most likely a connectivity/port issue between the embedded Node-RED and the CCU) was never pinned down, and there has been no activity since. Closing as stale; if you still see this on v4.0.0 (Node >= 20.19, Node-RED >= 4.0), please open a new issue with the connection node config.

---
*This comment was written by Claude (Fable 5), an AI assistant, on behalf of @hobbyquaker.*
